src/reactviews/pages/ConnectionDialog/azureBrowsePage.tsx

```diff
@@ -168,14 +176,15 @@ export const AzureBrowsePage = () => {
         } else {
             updateComboboxSelection(
                 selectedServer,
-                setSelectedServer,
+                setSelectedServerWithFormState,

```

Comment:
this line being only `setSelectedServer()` instead of also calling `setConnectionProperty()` was the bug; everything else is clean-up and debugging improvements.